src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs

```diff
@@ -472,6 +474,10 @@ public async Task GetStreamCapacity_SumInvariant()
 
             (QuicConnection clientConnection, QuicConnection serverConnection) = await CreateConnectedQuicConnection(clientOptions, listenerOptions);
 
+            while (maxStreamIndex < Limit)
+            {
+                await streamsAvailableFired.WaitAsync(PassingTestTimeout);
```

Comment:
SemaphoreSlim.WaitAsync(timeout) will wait until timeout and return false. So this may turn into an infinite loop.

Reply:
Isn't it going to throw an exception and that should cause failure on test?

Reply:
Ah, yes you're right, this is actually calling WaitAsync from `SemaphoreSlim` not from a task.

Reply:
Fixed.